components/menu/menu.tsx

```diff
@@ -0,0 +1,259 @@
+    const [disableAutoFocusItem, setDisableAutoFocusItem] = useState(false);
+    const isMenuOpen = Boolean(anchorElement);
+
+    function handleMenuClose(event: MouseEvent<HTMLDivElement>) {
+        event.preventDefault();
+        setAnchorElement(null);
+        setDisableAutoFocusItem(false);
+    }
+
+    function handleMenuClick() {
+        setAnchorElement(null);
+    }
+
+    function handleMenuKeyDown(event: KeyboardEvent<HTMLDivElement>) {
+        if (event.key === 'Enter') {
+            const target = event.target as HTMLElement;
+            const ariaHasPopup = target?.getAttribute('aria-haspopup') === 'true' ?? false;
+
+            // Avoid closing the sub menu item on enter
+            if (!ariaHasPopup) {
+                setAnchorElement(null);
+            }
+        }
+    }
+
+    function handleMenuButtonClick(event: SyntheticEvent<HTMLButtonElement>) {
+        event.preventDefault();
+
+        if (isMobileView) {
+            dispatch(
+                openModal<MenuModalProps>({
+                    modalId: `${props.menuButton.id}-${props.menu.id}`,
+                    dialogType: MenuModal,
+                    dialogProps: {
+                        menuButtonId: props.menuButton.id,
+                        menuId: props.menu.id,
+                        menuAriaLabel: props.menu?.['aria-label'] ?? '',
+                        children: props.children,
+                    },
+                }),
+            );
+        } else {
+            setAnchorElement(event.currentTarget);
+        }
+    }
+
+    function handleMenuButtonMouseDown() {
+        // This is needed to prevent focus-visible being set on clicking menuitems with mouse
+        setDisableAutoFocusItem(true);
+    }
+
+    function renderMenuButton() {
+        // We construct the menu button so we can set onClick correctly here to support both web and mobile view
+        const triggerElement = (
+            <button
+                id={props.menuButton.id}
+                aria-controls={props.menu.id}
+                aria-haspopup={true}
+                aria-expanded={isMenuOpen}
+                aria-label={props.menuButton?.['aria-label'] ?? ''}
+                className={props.menuButton?.class ?? ''}
+                onClick={handleMenuButtonClick}
+                onMouseDown={handleMenuButtonMouseDown}
+            >
+                {props.menuButton.children}
+            </button>
+        );
+
+        if (props.menuButtonTooltip && props.menuButtonTooltip.text && !isMobileView) {
+            return (
+                <OverlayTrigger
+                    delayShow={OVERLAY_TIME_DELAY}
+                    placement={props?.menuButtonTooltip?.placement ?? 'top'}
+                    overlay={
+                        <Tooltip
+                            id={props.menuButtonTooltip.id}
+                            className={props.menuButtonTooltip?.class ?? ''}
```

Comment:
`??` not needed here

Reply:
i didnt want it to have undefined in className hence ??

Reply:
Whether an empty string or undefined, it has the same outcome.